oxy/poc_translator.py

```diff
@@ -0,0 +1,73 @@
+from ruamel.yaml import YAML
+from functools import reduce
+
+
+yaml: YAML = YAML(typ="safe")
+
+'''
+pub fn decode_accumulator_status(data: &[u8]) -> HashMap<u8, f32> {
+    let mut result = HashMap::new();
+    result.insert(1, (pd::big_endian(&data[0..2], 8) as f32) / 10.0);
+    result.insert(
+        2,
+        pd::twos_comp(pd::big_endian(&data[2..4], 8) as u32, 16) as f32 / 10.0,
+    );
+    result.insert(3, pd::big_endian(&data[4..6], 8) as f32);
+    result.insert(4, data[6] as f32);
+    result.insert(5, data[7] as f32);
+    result
+}
+'''
+out_string: str = ""
+data = yaml.load(open("mapping.yaml"))[1]
+# print(data)
+
+function_name: str = "decode" + "_" + "_".join(data['string'].split(" "))
+args: str = "(data: &[u8])"
+returnVal: str= " -> HashMap<u8, f32>"
+
+signature: str = "pub fn " + function_name + args + returnVal + " {"
+instantiate_hash_map: str = "    let mut result = HashMap::new();"
+conclusion: str = "    result\n}"
+
+decodings: list[str] = []
+accumulated_size: int = 0
+for field in data["fields"]: # result.insert(1, (pd::big_endian(&data[0..2], 8) as f32) / 10.0);
+    field: dict = field
+    id = field["field_id"]
+    if field["size"] > 1: # we need to do some decoding, then
+        to_decode: str = f"&data[{accumulated_size}..{accumulated_size+field['size']}]"
+        _cf: str = field.get("correcting_factor", "")
+        correcting_factor: str = f"{(' / ' if '/' in _cf else ' * ') if _cf else ''}{_cf.split('/')[-1]}"
+        for decodingsetup in field["decoding"]:
+            decodingsetup: dict[str, dict[str, str]] = {k: reduce(lambda x,y: x|y, v, {}) for k,v in decodingsetup.items()}
+            for decoder, params in decodingsetup.items():
+                match decoder:
+                    case "big_endian":
```

Comment:
also if ur gonna do this just u dont even need the match technically just do f”pd::{decoder}({to_decode}, {bits}) {final_type}

Reply:
just rename function names to match this in the rust code

Reply:
or change yaml to match function names